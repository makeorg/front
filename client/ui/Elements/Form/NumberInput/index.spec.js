import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { NumberInput } from './index';

jest.mock('Client/ui/Elements/Form/Styled/Input', () => ({
  BasicInputStyle: 'BasicInputStyle',
}));

jest.mock('Client/ui/Elements/Form/Styled/Content', () => ({
  MiddleFakeFieldStyle: 'MiddleFakeFieldStyle',
  FloatingLabelStyle: 'FloatingLabelStyle',
  FieldWrapperStyle: 'FieldWrapperStyle',
}));

jest.mock('Client/ui/Elements/Form/Styled/Icons', () => ({
  CenterInputIconStyle: 'CenterInputIconStyle',
}));

const defaultProps = {
  type: 'foo',
  name: 'bar',
  icon: 'SvgEnvelope',
  value: 'baz',
  label: 'qux',
  handleChange: () => {},
};

describe('NumberInput', () => {
  it('must return the diff between snapshot when input is required or optionnal', () => {
    const RequiredInput = renderer
      .create(<NumberInput {...defaultProps} required />)
      .toJSON();
    const OptionnalInput = renderer
      .create(<NumberInput {...defaultProps} />)
      .toJSON();
    expect(snapshotDiff(RequiredInput, OptionnalInput)).toMatchSnapshot();
  });
});
