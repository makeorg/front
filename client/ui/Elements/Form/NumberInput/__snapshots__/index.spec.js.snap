// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput must return the diff between snapshot when input is required or optionnal 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -12,11 +12,11 @@
        id="bar"
        max={100000}
        min={0}
        name="bar"
        onChange={[Function anonymous]}
-       required={true}
+       required={false}
        type="number"
        value="baz"
      />
      <FloatingLabelStyle
        htmlFor="bar"
`;
